### ISSUE TITLE
goreturns is not formatting on save, gofmt, and goimports work

### ISSUE THREAD

**Issue by josvazg** [Microsoft/vscode-go#1821]
goreturns is not working for me on save with the vs-code go plugin. Changing "go.formatTool" to gofmt or goimports works.

All three programs are reachable in the path, and they seem to work fine when invoked manually. For instance goreturns even works with the -d option.

I do not have a "go.useDiffForFormatting" option anymore, that seemed to be the fix for this or a similar issue before.

I am running Linux Ubuntu 18.04, VS Code 1.25.1 and the plugin is Go 0.6.85

**Comment by ramya-rao-a** [comment]
- Does `goreturns` work when you manually format the file in VS Code i.e right-click -> Format Document?
- Are there any errors in the console? Help -> Toggle Developer Tools -> Console
- Can you try updating the tool? Run `Go: Install/Update Tools`, select `goreturns` to update

**Comment by josvazg** [comment]
Weird, it got fixed. I was using goimports so my code was formatted properly beforehand.

I changed my settings back to goreturns and made a change to break the format. I didn't even get to use the formatting option in the right-click menu, tried to save and vscode prompted me to update goreturns.

That was the first time I saw a reaction from vscode on saving whie having goreturns setup.

Once I accepted the install goreturns started working as expected on save.
It also works on the right-click Format Document.
(Now) I see no errors in the console. Nice to learn about that BTW.

Thanks! I will close this now.
